CONTRIBUTING.md

```diff
@@ -0,0 +1,70 @@
+# (WIP) Contributing Info 
+
+## Semantic Commit Messages
+
+Inspired by [Git Commit Msg][git-commit-msg], [Angular Commit Message Format][angular-commit-message-format] and [Conventional Commits][conventional-commits]
+
+
+### <a name="commit-message-format"></a> Commit Message Format
+
+Each commit message consists of a **header**, a **body** and a **footer**. The header has a special
+format that includes a **type**, a **scope**, and a **subject**:
+
+```html
+<type>(<scope>): <subject>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+
+> Any line of the commit message cannot be longer 100 characters!<br/>
+This allows the message to be easier to read on GitHub as well as in various Git tools.
+
+##### Type
+
+Must be one of the following:
+
+* **feat**: A new feature
+* **fix**: A bug fix
+* **docs**: Documentation only changes
+* **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing
+  semi-colons, etc)
+* **refactor**: A code change that neither fixes a bug nor adds a feature
+* **perf**: A code change that improves performance
+* **test**: Adding missing tests
+* **chore**: Changes to the auxiliary tools such as release scripts
+* **build**: Changes to the dependencies, devDependencies, or build tooling
+* **ci**: Changes to our Continuous Integration configuration
+
+##### Scope
+
+The scope could be anything that helps specify the scope (or feature) that is changing.
```

Comment:
Should the scope be optional?

Reply:
Yes scope is optional. I will change the text a little bit